### PR TITLE
[8.13.x] Bump Quarkus version to 2.13.4.Final (#453)

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/2820
- https://github.com/kiegroup/kogito-apps/pull/1639
- https://github.com/kiegroup/kogito-examples/pull/1581
- https://github.com/kiegroup/optaplanner/pull/2639
- https://github.com/kiegroup/optaplanner-quickstarts/pull/522

### CODE REVIEW

File: .github/workflows/full-downstream.yml [hunk 1]
```diff
@@ -50,10 +50,11 @@ jobs:
           restore-keys: ${{ runner.os }}-${{ matrix.java-version }}-m2
       - name: Build Chain ${{ matrix.java-version }}
         id: build-chain
-        uses: kiegroup/github-action-build-chain@v2.6.8
+        uses: kiegroup/kogito-pipelines/.ci/actions/build-chain@1.13.x
         with:
-          definition-file: https://raw.githubusercontent.com/${GROUP}/kogito-pipelines/main/.ci/pull-request-config.yaml
+          annotations-prefix: ${{ runner.os }}-${{ matrix.java-version }}
```
Comment (radtriste) [inline .github/workflows/full-downstream.yml:56]:
```suggestion
          definition-file: https://raw.githubusercontent.com/kiegroup/kogito-pipelines/1.13.x/.ci/pull-request-config.yaml
          annotations-prefix: ${{ runner.os }}-${{ matrix.java-version }}
```

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -40,11 +40,10 @@ jobs:
       - name: Build Chain
         uses: kiegroup/kogito-pipelines/.ci/actions/build-chain@1.13.x
         with:
-          definition-file: https://raw.githubusercontent.com/kiegroup/kogito-pipelines/%{process.env.GITHUB_BASE_REF.replace(/(\d*)\.(.*)\.(.*)/g, (m, n1, n2, n3) => `${+n1-7}.${n2}.${n3}`).replace('development', 'main')}/.ci/pull-request-config.yaml
           annotations-prefix: ${{ runner.os }}-${{ matrix.java-version }}/${{ matrix.maven-version }}
```
Comment (radtriste) [inline .github/workflows/pull_request.yml:44]:
```suggestion
          definition-file: https://raw.githubusercontent.com/kiegroup/kogito-pipelines/1.13.x/.ci/pull-request-config.yaml
          annotations-prefix: ${{ runner.os }}-${{ matrix.java-version }}/${{ matrix.maven-version }}
```

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by radtriste — APPROVED** [review]

**Review by rsynek — APPROVED** [review]

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Pull request merged**